### ISSUE TITLE
Run a function when socket disconnects (Doesn't work inside function close())

### ISSUE THREAD

**Issue by Ogarunite** [OgarProject/Ogar#541]
TLDR: How can i run a specific function when someone disconnects?

Hi, i'm trying to run a specific function when someone disconnects from the server.

So when someone disconnects, there is the function close(error) inside GameServer.js.

So I want to run a function inside function close(error), or even the function content inside function close(error) but i keep getting the error "cannot read property "length" of undefined.

This is how it looks like:
`function` `close(error)` `{`
       `  `   `console.log("[Game]` `Disconnect:` `"+error);```
           
            //Loop i want to do
              var clientz = this.clients.length;
              var countz = 0;
               for(var i=0; i<clientzz; i++) {
                 console.log('something, or something like remaining clients');
                countz++;
              }
              console.log(countz);
          };
`

But i keep getting that error...

**Comment by Andrews54757** [comment]
take a look at my code

**Comment by Ogarunite** [comment]
I took a look at your function close(error), the borders+ upon disconnect was interesting. But I still couldn't find a solution to run a code when someone disconnects

**Comment by NatsuTheGreat** [comment]
Dude you make a very simple mistake:

You said the error is it can't read the length of undefined (reference error)

I see the code you defined the variable *clientz* but in the loop you used *clientzz* , client was never defined, that's why you get that error.

Replace:

````
          for(var i=0; i<clientzz; i++) {
````

With:

````
          for(var i=0; i<clientz; i++) {
````

Tell me if something is still wrong.

**Comment by Andrews54757** [comment]
@NatsuTheGreat It still wouldnt work because `this` would be a different thing. I ran into this problem in my code and I solved it by first defining `gameServern = this` in the start function as a global variable then by using it in the disconnect function and it would work so instead of this, you would do gameServern

**Comment by NatsuTheGreat** [comment]
Oh, i've never tested his code, I just wanted to solve his problem with the undefined variable part.

**Comment by ItzLevvie** [comment]
Close? @vram4

**Comment by Ogarunite** [comment]
Yes! Thank you very much @Andrews54757 it worked! 

Closable!
